packages/opentelemetry-tracing/src/config.ts

```diff
@@ -32,7 +32,7 @@ export const DEFAULT_MAX_LINKS_PER_SPAN = 32;
 export const DEFAULT_CONFIG = {
   defaultAttributes: {},
   logLevel: LogLevel.INFO,
-  sampler: ALWAYS_SAMPLER,
+  sampler: new ProbabilitySampler(Number(process.env.OTEL_SAMPLING_RATE || 1)),
```

Comment:
This config is created when this file is required so well before your test tries to set it into the environment.
I don't think you should change the default config. I would have checked the existence of the env variable inside the BasicTracer and override the sampler there. Also you'll like the other PR to have a different implementation for the browser since `process` will not be available

Reply:
> This config is created when this file is required so well before your test tries to set it into the environment.
> I don't think you should change the default config. I would have checked the existence of the env variable inside the BasicTracer and override the sampler there. 

Also tried this approach but the env var is not working as expected 🤔 

```
  1) BasicTracerProvider
       .startSpan()
         should not sample a trace when OTEL_SAMPLING_RATE is 0:

      AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- 1
+ 0
```

Reply:
I don't know then, maybe try to use the `ConsoleLogger` and increase the verbosity to debug ?

Reply:
Hmmm in the end I only had to make changes to `Tracer.ts` but not `BasicTracerProvider,ts` 🤷‍♂️